src/rtctools/optimization/multi_seed_mixin.py

```diff
@@ -0,0 +1,48 @@
+from typing import Dict, Iterable, Optional
+
+from rtctools.optimization.optimization_problem import OptimizationProblem
+from rtctools.optimization.timeseries import Timeseries
+
+
+class MultiSeedMixin(OptimizationProblem):
+    """
+    Enables a workflow to solve an optimization problem by trying multiple seeds.
+    """
+
+    def __init__(self, **kwargs):
+        self.__selected_seed = None
+        super().__init__(**kwargs)
+
+    def seeds(self) -> Iterable[Optional[Dict[str, Timeseries]]]:
+        """Return a list or iterator of seeds.
+
+        When the seed is None, the default seed will be used.
+        This method should be implemented by the user.
+        """
```

Comment:
Yes, this is much cleaner! Thank you for taking the time to refactor this.

Can we 'raise NotImplementedError' here by default, rather than returning the somewhat artificial '[None]' list?
If this method is not overridden/implemented, the user should not be using the mixin.

Reply:
I agree this is a lot leaner, but I feel that now the user has to do too much himself if a parameter like theta_start depends on the seed. On the other hand, having these solver parameters depend on the seed also doesn't sound very logical. I think I'll try one more thing to see if we can use multiple seeds with different solver parameters in a clean way.

Reply:
I've created an alternative that fits the workflow that we had in mind a bit better: #1645.